### PR TITLE
gha update dist folder on release #3547

### LINKED ISSUE #3547: Github Scan Action: Update `dist/` folder on release
## Situation
After trying hard to eliminate the `dist/` folder containing e.g. the generated `index.js`
we realized that we face too many side effects as well as performance issues when running the action.

## ToDo
When running the release workflow `https://github.com/mercedes-benz/sechub/actions/workflows/release-github-action.yml`
then make sure that the contents of the `dist/` folder are being
* generated
* and added to the "1 - Before Github Action release: Merge release artifacts [auto-generated]" pull request

### PR DESCRIPTION
**Pull request by sven-dmlr**
* add dist/ folder to PR on release
* made SecHub client caching work
* cleanup composite steps
* removed branch parameter
* no more git checkout of action - use ${{github.action_path}} instead
* use a defined nodejs version (also hash instead of version tag)
* integrationtests: use latest server and pds versions
* closes #3547

### CODE REVIEW

**Review by hamidonos — COMMENTED** [review]
branch name is wrong

for gh actions we use "gha_feature-......"

**Comment by sven-dmlr** [comment]
> branch name is wrong
> 
> for gh actions we use "gha_feature-......"

As per https://mercedes-benz.github.io/sechub/latest/sechub-techdoc.html#branching-model this rule is deprecated :-)

File: .github/workflows/github-action-scan.yml [hunk 1]
```diff
@@ -54,9 +54,9 @@ jobs:
       - name: Define integration test setup
         id : version-selector
         run: |
-          echo "sechub_server_version=1.9.0" >> "$GITHUB_ENV"
+          echo "sechub_server_version=2.3.1" >> "$GITHUB_ENV"
           echo "sechub_server_port=8443" >> "$GITHUB_ENV"
-          echo "pds_version=1.6.0" >> "$GITHUB_ENV"
+          echo "pds_version=2.0.0" >> "$GITHUB_ENV"
```
[the remark this answers is not in the document]
Comment (sven-dmlr) [reply .github/workflows/github-action-scan.yml]: done

**Review by hamidonos — APPROVED** [review]
good job :) lets merge it

**Pull request merged**